internal/rpc/jsonrpc/methods.go

```diff
@@ -2400,23 +2405,67 @@ func (s *Server) purchaseTicket(ctx context.Context, icmd interface{}) (interfac
 		expiry = int32(*cmd.Expiry)
 	}
 
-	ticketFee := w.TicketFeeIncrement()
+	// The ticketFee code is being deprecated, but still need to be here because of
+	// positional json-rpc parameters.
 
 	// Set the ticket fee if specified.
 	if cmd.TicketFee != nil {
-		ticketFee, err = dcrutil.NewAmount(*cmd.TicketFee)
+		_, err = dcrutil.NewAmount(*cmd.TicketFee)
 		if err != nil {
 			return nil, rpcError(dcrjson.ErrRPCInvalidParameter, err)
 		}
 	}
 
-	hashes, err := w.PurchaseTickets(ctx, 0, spendLimit, minConf, ticketAddr,
-		account, numTickets, poolAddr, poolFee, expiry, w.RelayFee(),
-		ticketFee)
+	dontSignTx := false
+	if cmd.DontSignTx != nil {
+		dontSignTx = *cmd.DontSignTx
+	}
+
+	request := &wallet.PurchaseTicketsRequest{
+		Count:         numTickets,
+		SourceAccount: account,
+		VotingAddress: ticketAddr,
+		MinConf:       minConf,
+		Expiry:        expiry,
+		DontSignTx:    dontSignTx,
+		VSPAddress:    poolAddr,
+		VSPFees:       poolFee,
+	}
+	ticketsResponse, err := w.PurchaseTicketsWithResponse(ctx, n, request)
 	if err != nil {
 		return nil, err
 	}
+	ticketsTx := ticketsResponse.Tickets
+	splitTx := ticketsResponse.SplitTx
+
+	// If we dont need to sign the transaction we return its unsigned tickets
+	// bytes and the splittx. So a cold wallet can handle it.
+
+	var stringBuilder strings.Builder
```

Comment:
This shared builder is causing issues with the returned results.  When the builder is reused to serialize a second transaction, the hex encoding of the transaction is simply concatenated to the previous transaction.

You need to either use separate builder variables, or call its Reset method before each serialize.